### ISSUE TITLE
Convert SPARQLWrapper.SmartWrapper to pandas dataframe

### ISSUE THREAD

**Issue by appliedgraphs** [RDFLib/sparqlwrapper#205]
I want to convert a SmartWrapper to a pandas DF.  We have two relevant issues: 

- [#125](https://github.com/RDFLib/sparqlwrapper/issues/125) Which is focused on a SPARQLResult
- [#1179](https://github.com/RDFLib/rdflib/issues/1179) - which also seems to be about SPARQLRest

My goal is to execute a SPARQL query, get a result, and then transform to a dataframe.  I am certain that I am missing something because this must be a very common need.

**Comment by nicholascar** [comment]
@appliedgraphs have you seen this exsiting script: https://github.com/RDFLib/sparqlwrapper/blob/master/SPARQLWrapper/sparql_dataframe.py

**Comment by afk314** [comment]
I hadn't, but it's nice and easy to follow.  I will try it out later today.  Thanks.

Could something like this be folded into rdflib/sparqlwrapper so that it was at hand.  I'm optimistic more and more folks will want to play with graph ml approaches and going to pandas is just the default at this point.

**Comment by nicholascar** [comment]
Yes, I think PANDAS is well known enough to be supported as a standard output option. The only hitch is that we want to ensure basic installation of SPARQLWrapper doesn't _require_ PANDAS, since it's large, so it can be included but a pip install optional will need to be made to enable it.

Something like `pandas[pandas]` in requirements.txt and then `pip install sparqlwrapper[pandas]` to install.

Would love to see a PR for this...

**Comment by eggplants** [comment]
@nicholascar Already:
https://github.com/RDFLib/sparqlwrapper/blob/2a6e2d3ddbc3fe38ca47d6d05f23c9b61ff82366/setup.cfg#L53-L54

**Comment by nicholascar** [comment]
Great, thanks @eggplants, so the installation tasks are already done. So, I suppose, a deeper integration of PANDAS could be made, as long as appropriate warnings are thrown if users try and call PANDAS exports without the necessary installation.
